controllers/disruption_controller.go

```diff
@@ -152,6 +158,11 @@ func (r *DisruptionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 		// the injection is being created or modified, apply needed actions
 		controllerutil.AddFinalizer(instance, chaostypes.DisruptionFinalizer)
+		if err := r.Update(context.Background(), instance); err != nil {
```

Comment:
this was the source of the orphans

Reply:
Should we keep the `handleOrphanedChaosPods` function now?

Reply:
Though it means that there's still the case of someone editing the disruption to manually remove the finalizer and deleting the disruption while there are running chaos pods, right?
If yes, what do you think to implement (in another PR of course, eventually later) a mechanism where the admission webhook would reject a disruption update removing the finalizer if there are existing chaos pods for the related disruption? It would prevent such a scenario + it would probably be helpful as it forces the user to check for failed pods in stuck on removal cases?

Reply:
> If yes, what do you think to implement (in another PR of course, eventually later) a mechanism where the admission webhook would reject a disruption update removing the finalizer if there are existing chaos pods for the related disruption? It would prevent such a scenario + it would probably be helpful as it forces the user to check for failed pods in stuck on removal cases?

I think that's great and I'll file a ticket for it